kqueen/engines/openstack_kubespray.py

```diff
@@ -477,18 +461,31 @@ def _save_inventory(self, inventory, filename):
         with open(self._get_cluster_path(filename), "w") as fp:
             json.dump(inventory, fp, indent=4)
 
-    def _create_group_vars(self):
+    def _create_group_vars(self, metadata):
         src = os.path.join(self.kubespray_path, "inventory/sample/group_vars")
         dst = self._get_cluster_path("group_vars")
         shutil.copytree(src, dst)
 
-        with open(os.path.join(dst, "all.yml"), "a") as all_yaml:
-            data_to_add = {'cloud_provider': 'openstack'}
-            image_var_names = [var_name for var_name in dir(config) if var_name.endswith(('_IMAGE_REPO', '_IMAGE_TAG'))]
-            image_variables = {k.lower(): getattr(config, k) for k in image_var_names}
-            data_to_add.update(image_variables)
+        persistent_volumes = metadata.get("persistent_volume")
+        kubespray_vars = {"persistent_volumes_enabled": True} if persistent_volumes else {}
```

Comment:
``kubespray_vars = {"persistent_volumes_enabled": True} if metadata.get("persistent_volume") else {}``
or i missed , that u need this var?

Reply:
This far is responsible for storage class, other ones for the whole deployment (set in a separate commit)